src/diffusers/workflow_utils.py

```diff
@@ -0,0 +1,135 @@
+        pipeline_name_or_path (`str`): Name or the local path to the pipeline that was used to generate the workflow.
+
+    Returns:
+        `Dict`: A dictionary containing the details of the pipeline call arguments and (optionally) LoRA checkpoint
+        details.
+    """
+    workflow = Workflow()
+
+    # Populate call arguments.
+    call_arguments = {
+        arg: call_arg_values[arg]
+        for arg in argument_names
+        if arg != "return_workflow"
+        and not isinstance(call_arg_values[arg], (torch.Tensor, np.ndarray, PIL.Image.Image, Callable))
+    }
+
+    workflow.update(call_arguments)
+
+    generator = workflow.pop("generator")
+    if generator is not None:
+        try:
+            workflow.update({"generator_seed": generator.initial_seed()})
+            workflow.update({"generator_device": generator.device})
+        except Exception:
+            workflow.update({"generator_seed": None})
+            workflow.update({"generator_device": "cpu"})
+
+    workflow["_name_or_path"] = pipeline_name_or_path
+
+    return workflow
+
+
+class Workflow(dict, ConfigMixin, PushToHubMixin):
+    """Class sub-classing from native Python dict to have support for interacting with the Hub."""
+
+    config_name = None
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config_name = WORKFLOW_NAME
+        self._internal_dict = {}
+
+    def __setitem__(self, __key, __value):
+        self._internal_dict[__key] = __value
+        return super().__setitem__(__key, __value)
+
+    def update(self, __m, **kwargs):
+        self._internal_dict.update(__m, **kwargs)
+        super().update(__m, **kwargs)
+
+    def pop(self, __key):
+        self._internal_dict.pop(__key)
+        super().pop(__key)
+
+    def save_workflow(self, save_directory: Union[str, os.PathLike], push_to_hub: bool = False, filename: str = WORKFLOW_NAME):
+        """
+        Saves a workflow to a directory.
+
+        Args:
+            save_directory (`str` or `os.PathLike`):
+                Directory where the workflow JSON file will be saved (will be created if it does not exist).
+            push_to_hub (`bool`, *optional*, defaults to `False`):
+                Whether or not to push your model to the Hugging Face Hub after saving it. You can specify the
+                repository you want to push to with `repo_id` (will default to the name of `save_directory` in your
+                namespace).
+            filename (`str`, *optional*, defaults to `workflow.json`):
+                Optional filename to use to serialize the workflow JSON.
+        """
+        self.config_name = filename
+        self.save_config(save_directory=save_directory, push_to_hub=push_to_hub)
+
+    def save_pretrained(
+        self,
+        save_directory: Union[str, os.PathLike],
+        push_to_hub: bool = False,
+        filename: str = WORKFLOW_NAME,
+        **kwargs,
```

Comment:
@patrickvonplaten the reason why we have it here is to support `push_to_hub()` directly on a workflow object. Otherwise, it will fail complaining about `safe_serialization` wasn't passed. 

https://github.com/huggingface/diffusers/blob/f912f39b50f87e50a9d99346f5c1b6e644653262/src/diffusers/utils/hub_utils.py#L451